datafusion/physical-expr/src/expressions/dynamic_filters.rs

```diff
@@ -105,47 +97,44 @@ impl DynamicFilterPhysicalExpr {
         inner: Arc<dyn PhysicalExpr>,
     ) -> Self {
         Self {
-            // columns: children,
-            // remapped_columns: None, // Initially no remapped children
-            remapped_schema: None,
-            // remapped_filter: None,
-            inner: Arc::new(RwLock::new(inner)),
+            inner,
             data_type: Arc::new(RwLock::new(None)),
             nullable: Arc::new(RwLock::new(None)),
         }
     }
 
     // udpate schema
-    pub fn with_schema(
-        &self,
-        schema: SchemaRef,
-    ) -> Self {
-        Self {
-            remapped_schema: Some(schema),
-            inner: Arc::clone(&self.inner),
-            data_type: Arc::clone(&self.data_type),
-            nullable: Arc::clone(&self.nullable),
-        }
-    }
+    // pub fn with_schema(&self, schema: SchemaRef) -> Self {
+    //     Self {
+    //         remapped_schema: Some(schema),
+    //         inner: Arc::clone(&self.inner),
+    //         data_type: Arc::clone(&self.data_type),
+    //         nullable: Arc::clone(&self.nullable),
+    //     }
+    // }
 
     // get the source filter
     pub fn current(&self) -> Result<Arc<dyn PhysicalExpr>> {
-        let inner = self
-            .inner
-            .read()
-            .map_err(|_| {
-                datafusion_common::DataFusionError::Execution(
-                    "Failed to acquire read lock for inner".to_string(),
-                )
-            })?
-            .clone();
+        let inner = Arc::clone(&self.inner);
+
+        // let inner = self
+        //     .inner
+        //     .read()
+        //     .map_err(|_| {
+        //         datafusion_common::DataFusionError::Execution(
+        //             "Failed to acquire read lock for inner".to_string(),
+        //         )
+        //     })?
+        //     .clone();
         Ok(inner)
     }
 
     // update source filter
-    pub fn update(&self, filter: PhysicalExprRef) {
-        let mut w = self.inner.write().unwrap();
-        *w = filter;
+    // create a new one
+    pub fn update(&mut self, filter: PhysicalExprRef) {
+        self.inner = filter;
```

Comment:
How will writers have mutable access to this if they have to package it up in an `Arc`?